### PR TITLE
Fixes npc not showing after going out of range

### LINKED ISSUE #6: Replacing spigot causes npc to disappear.
When we changed our server core from PaperSpigot to PandaSpigot, the npc of the Citizens plugin disappeared.
Citizens version: v2.0.16-SNAPSHOT(build 1247)

Before replacing spigot.

After replacing spigot.

### LINKED ISSUE #27: Chunk render bug
Happens with hologramsm with players and even with NPC not being perfectly rendered, when you unrender and render from client-side, they look like this

### LINKED ISSUE #118: NPCs don't like PandaSpigot
Steps to craft:
1- Create an NPC using Citizens 
2- Go away from the NPC (until you don't see the NPC anymore) 
3- Go back to where the NPC is and he will be gone

Note: I tested the same server with Paper and it works normally
Video: https://www.youtube.com/watch?v=k2f-pWgpEUA
My Server Test: https://www.mediafire.com/file/hjr2c1gov7mpkaa/factions+-+Copia.zip/file

Potentially reported on: #6 , #27

### PR DESCRIPTION
**Pull request by mechoriet**
solves a problem when npc's get out of sight and go back to it they don't become visible again
and probably also solves other issues related to entity tracking

closes #118 
closes #6 
Fixes #27

### CODE REVIEW

**Comment by mechoriet** [comment]
can we checked think I've have done most of clean up

File: patches/server/0009-Optimize-Network-Manager-and-add-advanced-packet-sup.patch [hunk 1]
```diff
@@ -214,36 +207,36 @@ index 8f8b9aa731f2c97a3e32369bfd4d641a1f2ab412..f6e797f712f029a9bbe94739469c7464
...
++                // PandaSpigot end
                      channelfuture.addListener(ChannelFutureListener.FIRE_EXCEPTION_ON_FAILURE);
-+                    // PandaSpigot start
-+                    } catch (Exception e) {
-+                        g.error("NetworkException: " + player, e);
-+                        close(new ChatMessage("disconnect.genericReason", "Internal Exception: " + e.getMessage()));;
-+                        packet.onPacketDispatchFinish(player, null);
-+                    }
-+                    // PandaSpigot end
++                // PandaSpigot start
```
Comment (uRyanxD) [inline patches/server/0009-Optimize-Network-Manager-and-add-advanced-packet-sup.patch:233]: method/comments codestyle issue

File: patches/server/0009-Optimize-Network-Manager-and-add-advanced-packet-sup.patch [hunk 2]
```diff
@@ -285,117 +278,100 @@ index 8f8b9aa731f2c97a3e32369bfd4d641a1f2ab412..f6e797f712f029a9bbe94739469c7464
...
-index fdc142b75b78289f01a9e9fed6f36432624735db..d60ec06b995eb477b555d222e99e208560656cef 100644
+index fdc142b75b78289f01a9e9fed6f36432624735db..8c9cd810779d1bbffb94dcab641602c05d19182d 100644
 --- a/src/main/java/net/minecraft/server/Packet.java
 +++ b/src/main/java/net/minecraft/server/Packet.java
-@@ -9,4 +9,37 @@ public interface Packet<T extends PacketListener> {
+@@ -9,4 +9,19 @@ public interface Packet<T extends PacketListener> {
      void b(PacketDataSerializer packetdataserializer) throws IOException;
  
      void a(T t0);
-+
```
Comment (uRyanxD) [inline patches/server/0009-Optimize-Network-Manager-and-add-advanced-packet-sup.patch:367]: Maybe revert those changes in the docs? The one from hpxfd looks a little better to me

File: patches/server/0043-Add-Channel-initialization-listeners.patch [hunk 3]
```diff
@@ -5,15 +5,14 @@ Subject: [PATCH] Add Channel initialization listeners
...
+index b0067f486d59eb40d11128cf60d1844f38423e3e..8d09a48911368fb91e72c734d88c325078e89ea1 100644
 --- a/build.gradle.kts
 +++ b/build.gradle.kts
-@@ -41,6 +41,8 @@ dependencies {
+@@ -41,6 +41,7 @@ dependencies {
      implementation("org.jline:jline-terminal-jansi:3.20.0")
      // PandaSpigot end
  
 +    implementation("net.kyori:adventure-key:4.10.1") // PandaSpigot - Add Channel initialization listeners
-+
```
Comment (uRyanxD) [inline patches/server/0043-Add-Channel-initialization-listeners.patch:16]: why remove the space?

**Review by uRyanxD — CHANGES_REQUESTED** [review]

File: patches/server/0009-Optimize-Network-Manager-and-add-advanced-packet-sup.patch [hunk 4]
```diff
@@ -73,35 +72,33 @@ index 8f8b9aa731f2c97a3e32369bfd4d641a1f2ab412..f6e797f712f029a9bbe94739469c7464
...
 +        }
 +    }
-+    private static class InnerUtil { // Attempt to hide these methods from ProtocolLib, so it doesn't accidentally pick them up.
++    private static class InnerUtil { // Attempt to hide these methods from ProtocolLib so it doesn't accidently pick them up.
 +        private static java.util.List<Packet> buildExtraPackets(Packet packet) {
 +            java.util.List<Packet> extra = packet.getExtraPackets();
-+            if (extra == null || extra.isEmpty()) {
-+                return null;
-             }
++           if (extra == null || extra.isEmpty()) {
```
[the remark this answers is not in the document]
Comment (mechoriet) [reply patches/server/0009-Optimize-Network-Manager-and-add-advanced-packet-sup.patch:81]: looks fine to me 🤷 
```java
        private static java.util.List<Packet> buildExtraPackets(Packet packet) {
            java.util.List<Packet> extra = packet.getExtraPackets();
           if (extra == null || extra.isEmpty()) {
               return null;
            }
            java.util.List<Packet> ret = new java.util.ArrayList<>(1 + extra.size());
            buildExtraPackets0(extra, ret);
            return ret;
        }
```

File: patches/server/0013-Disable-Explicit-Network-Manager-Flushing.patch [hunk 5]
```diff
@@ -1,6 +1,6 @@
 From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
 From: Aikar <aikar@aikar.co>
-Date: Fri, 29 Oct 2021 05:24:16 -0400
+Date: Mon, 17 Apr 2023 23:49:40 +0200
```
Comment (uRyanxD) [inline patches/server/0013-Disable-Explicit-Network-Manager-Flushing.patch:3]: unnecessary date change

File: patches/server/0012-Optimise-non-flush-packet-sending.patch [hunk 6]
```diff
@@ -1,6 +1,6 @@
 From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
 From: Spottedleaf <spottedleaf@spottedleaf.dev>
-Date: Fri, 29 Oct 2021 08:25:50 -0400
+Date: Fri, 29 Oct 2021 05:24:16 -0400
```
Comment (uRyanxD) [inline patches/server/0012-Optimise-non-flush-packet-sending.patch:3]: unnecessary date change

File: patches/server/0011-Consolidate-flush-calls-for-entity-tracker-packets.patch [hunk 7]
```diff
@@ -24,9 +24,9 @@ index 35ed8ca8f5c40f73c0256b48c0350f437a140f01..c7b6bc4eb4103af9e3385b88984271a0
 +                    }
 +                }
 +                try {
-+                // PandaSpigot end
++                    // PandaSpigot end
```
Comment (uRyanxD) [inline patches/server/0011-Consolidate-flush-calls-for-entity-tracker-packets.patch:27]: comment codestyle

File: patches/server/0011-Consolidate-flush-calls-for-entity-tracker-packets.patch [hunk 8]
```diff
@@ -24,9 +24,9 @@ index 35ed8ca8f5c40f73c0256b48c0350f437a140f01..c7b6bc4eb4103af9e3385b88984271a0
...
                  worldserver.getTracker().updatePlayers();
-+                // PandaSpigot start - controlled flush for entity tracker packets
++                    // PandaSpigot start - controlled flush for entity tracker packets
```
Comment (uRyanxD) [inline patches/server/0011-Consolidate-flush-calls-for-entity-tracker-packets.patch:29]: comment codestyle

**Review by uRyanxD — APPROVED** [review]

**Pull request merged**